code-studio/src/js/clientState.js

```diff
@@ -251,9 +376,9 @@ function recordVisualElementSeen(visualElementType, visualElementId) {
  * @param visualElementId
  */
 function hasSeenVisualElement(visualElementType, visualElementId) {
-  var elementSeenJson = sessionStorage.getItem(visualElementType) || '{}';
+  var elementSeenJson = getItem(visualElementType) || {};
   try {
-    var elementSeen = JSON.parse(elementSeenJson);
+    var elementSeen = elementSeenJson;
     return elementSeen[visualElementId] === true;
```

Comment:
Three more above (`sourceForLevel`, `allLevelsProgress` and `recordVisualElementSeen`).